tracing-appender/src/rolling.rs

```diff
@@ -500,13 +507,60 @@ impl Inner {
                     .unwrap_or(0),
             ),
             rotation,
+            keep_last: None,
         };
         (inner, writer)
     }
 
+    fn prune_old_logs(&self, keep_last: usize) {
```

Comment:
If I understand this correctly, this function is collecting files into a `Vec` of `String`s if the path contains the filename prefix and then sorting by the dates.

While this can work, I don't think this is how this should be implemented. Namely, I suggest the following:
- [Use the path components](https://doc.rust-lang.org/stable/std/path/struct.Path.html#method.components) of each path and convert `log_file_name` to an `OsStr`.
- I don't recommend doing the adhoc string parsing in the latter half of the function. I believe we're emitting files with dates and I'm not opposed to enabling time's datetime parsing functionality in `tracing-appender` to better suppose this use-case, but depending on the compilation size hit, I might want to place this behind a Cargo feature.

Reply:
> [Use the path components](https://doc.rust-lang.org/stable/std/path/struct.Path.html#method.components) of each path and convert log_file_name to an OsStr.

instead of using `path components` and iterating till the last `Some(...)`, i think using the [file_name() from DirEntry](https://doc.rust-lang.org/std/fs/struct.DirEntry.html#method.file_name) can be more useful. It already returns an `OsStr` :)

Reply:
> I don't recommend doing the adhoc string parsing in the latter half of the function. I believe we're emitting files with dates and I'm not opposed to enabling time's datetime parsing functionality in tracing-appender to better suppose this use-case, but depending on the compilation size hit, I might want to place this behind a Cargo feature.

I implemented this, and pushed the change. However, I think it may be useful to compare two solutions. 
- 1st (adhoc): less code, same logic, but more error-prone since we are using some string slice, and if any mistake happens, it will be harder to discover and debug
- 2nd (date parsing): extra imports, an extra code block of 18 lines just to create a `formatter` for the date parser. Safer code, since we will be treating the string slice as a proper `Date` object.

*Important note:* we actually already calculated the `formatter` in the code. I wanted to use it via storing this in an extra field in the `Inner` struct. But the type of `formatter` is `FormatItem`, which is a lifetimed variable. So the `Inner` itself should have a lifetime, and I did not want to disrupt the clean/simple code. Hence, the code duplication in my solution. 
As you know, I'm not a mature Rust developer, but trying my best. If you think another solution may be better, please let me know. I'll try to implement it :)

Reply:
When you are happy with the solution, I can place all this behind a `feature` 👍